spice/spice.py

```diff
@@ -131,59 +127,63 @@ async def complete_response(self) -> SpiceResponse:
 class Spice:
     def __init__(
         self,
-        model: Optional[str] = None,
-        provider: Optional[str] = None,
-        model_aliases: Optional[Dict[str, Dict[str, str]]] = None,
+        default_text_model: Optional[TextModel | str] = None,
+        default_embeddings_model: Optional[EmbeddingModel | str] = None,
+        provider: Optional[Provider | str] = None,
```

Comment:
Been thinking about it and I think the provider should be given on a call by call basis, instead of at initialization. Since it's given at initialization, the user can't use 2 models from separate providers if one of them is an unknown model. Additionally, it just doesn't make sense to initialize for the client and be part of the client.